### PR TITLE
MPS Limits

### PR DESCRIPTION
**Pull request by yashrao99**
Not sure if the test file I added with it properly tests it... it passes the call of function for various values of the fault value

### CODE REVIEW

File: pcdsdevices/epics/mps.py [hunk 1]
```diff
@@ -134,3 +134,52 @@ def mps_factory(clsname, cls,  *args, mps_prefix, veto=False,  **kwargs):
     comp = FC(MPS, mps_prefix, veto=veto)
     cls  = type(clsname, (cls,), {'mps' : comp})
     return cls(*args, **kwargs)
+
+
+def mustBeOpenLogic(mps_A, mps_B):
```
Comment (teddyrendahl) [inline pcdsdevices/epics/mps.py]: Can you make these `must_be_open_logic`? We typically don't use camel case

File: pcdsdevices/epics/mps.py [hunk 2]
```diff
@@ -134,3 +134,52 @@ def mps_factory(clsname, cls,  *args, mps_prefix, veto=False,  **kwargs):
...
+        return False
+
+    if mps_A.fault.value == 1 and mps_B.fault.value == 0:
+        return False
+
+    if mps_A.fault.value == 0 and mps_B.fault.value == 1:
+        return True
+
+
+def mustKnowPositionLogic(mps_A, mps_B):
```
Comment (teddyrendahl) [inline pcdsdevices/epics/mps.py]: Also these functions should have docstrings

File: pcdsdevices/epics/mps.py [hunk 3]
```diff
@@ -134,3 +134,109 @@ def mps_factory(clsname, cls,  *args, mps_prefix, veto=False,  **kwargs):
...
+
+        self.MPSA = mps_A
+        self.MPSB = mps_B
+        self.name = name
+        self.logic = logic
+        super().__init__('', name=name)
+
+    @property
+    def faulted(self):
+
```
Comment (teddyrendahl) [inline pcdsdevices/epics/mps.py]: Add a docstring here that explains that we use the `logic` function to evaluate the faulted states of `mpsA` and `mpsB`

File: tests/test_mpslimits.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    mpsLimits.mps_A.bypass._read_pv.put(0)
+    assert mpsLimits.mps_A.faulted
+    
+    mpsLimits.mps_B.fault._read_pv.put(1)
+    mpsLimits.mps_B.bypass._read_pv.put(0)
+    assert mpsLimits.mps_B.faulted
+
+
+    mustBeOpenLogic(mpsLimits.mps_A, mpsLimits.mps_B)
+    mustKnowPositionLogic(mpsLimits.mps_A, mpsLimits.mps_B)
```
Comment (teddyrendahl) [inline tests/test_mpslimits.py]: I would put the straight logic test in their own function `test_logical_combinations` or something

File: tests/test_mpslimits.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+def test_mps_faults():
+    mpsLimits = fake_mps()
+    #Faulted
+    mpsLimits.mps_A.fault._read_pv.put(1)
+    mpsLimits.mps_A.bypass._read_pv.put(0)
+    assert mpsLimits.mps_A.faulted
+    
+    mpsLimits.mps_B.fault._read_pv.put(1)
+    mpsLimits.mps_B.bypass._read_pv.put(0)
+    assert mpsLimits.mps_B.faulted
```
Comment (teddyrendahl) [inline tests/test_mpslimits.py]: Yeah. Your intuition is right, this isn't quite testing what we want. You setup the problem right, but your `assert` statement doesn't check `mpsLimits.faulted`. What you really want to check is that `mpsLimits` returns the correct evaluation of the `logic` function for two active MPSLimits.

File: pcdsdevices/epics/mps.py (hunk 1 above)
Comment (ZLLentz) [reply pcdsdevices/epics/mps.py]: Chiming in to say that this is part of the pep8 specification for python style: https://www.python.org/dev/peps/pep-0008/#function-names

**Comment by teddyrendahl** [comment]
Nice work @yashrao99

**Pull request merged**